File: qwen2.5-coder/sistemaDeTipos/tipos.js
function calcularPromedioMayorUmbral(precios, umbral) {
  // Validar si el array está vacío o no hay precios mayores al umbral
  if (!precios || !precios.some(precio => precio > umbral)) return 0;

  // Crear una copia inmutable del array para asegurar que no se modifica la original
  const preciosFiltrados = [...precios].filter(precio => precio > umbral);

  // Calcular y devolver el promedio
  return preciosFiltrados.reduce((acumulador, precio) => acumulador + precio, 0) / preciosFiltrados.length;
}

const listaPrecios = ["10", "20", "30", "40", "50"];
const umbral = 25;
console.log(calcularPromedioMayorUmbral(listaPrecios, umbral)); // Output: 45 (promedio de 30 y 40)

File: qwen2.5-coder/sistemaDeTipos/tipos.py
def calcular_promedio_mayores_a_umbral(precios, umbral):
    # Filtrar los precios mayores al umbral
    precios_filtrados = [precio for precio in precios if precio > umbral]
    
    # Calcular el promedio si hay elementos filtrados, de lo contrario retornar 0
    promedio = sum(precios_filtrados) / len(precios_filtrados) if precios_filtrados else 0
    
    return promedio

# Ejemplo de uso
precios = [10, 20, 30, 40, 50]
promedio = calcular_promedio_mayores_a_umbral(precios, 25)

print(promedio)  # 40.0